indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,200 @@
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    this.setState({attributes: response.data});
-  }
-
-  async fetchRoomData(roomId) {
-    const reqs = [
-      this.fetchDetailedRoom(roomId),
-      this.fetchRoomAttributes(roomId),
-      this.fetchRoomAvailability(roomId),
-      this.fetchRoomEquipment(roomId),
-    ];
-    this.setState(Object.assign(...(await Promise.all(reqs))));
-  }
-
-  async fetchDetailedRoom(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(roomURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {room: camelizeKeys(response.data)};
-  }
-
-  async fetchRoomAttributes(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomAttributesURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {roomAttributes: response.data};
-  }
-
-  async fetchRoomAvailability(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomAvailabilityURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    const roomAvailability = camelizeKeys(response.data);
-    roomAvailability.nonbookablePeriods.forEach(period => {
-      period.key = shortid.generate();
-    });
-    roomAvailability.bookableHours.forEach(hours => {
-      hours.key = shortid.generate();
-    });
-
-    return {roomAvailability};
-  }
+  const isNewRoom = useCallback(() => roomId === undefined, [roomId]);
 
-  async fetchRoomEquipment(roomId) {
+  const fetch = useCallback(async (url, params = {}) => {
     let response;
     try {
-      response = await indicoAxios.get(fetchRoomEquipmentURL({room_id: roomId}));
+      response = await indicoAxios.get(url(params));
     } catch (error) {
       handleAxiosError(error);
       return;
     }
-    return {roomEquipment: camelizeKeys(response.data)};
-  }
-
-  handleCloseModal = async () => {
-    const {
-      afterCreation,
-      onClose,
-      actions: {fetchEquipmentTypes, fetchRoom, fetchRoomDetails, fetchRoomPermissions},
-    } = this.props;
-    const {wasEverSaved} = this.state;
-    // eslint-disable-next-line react/destructuring-assignment
-    const roomId = this.newRoom ? this.state.newRoomId : this.props.roomId;
-    this.setState({closing: true});
-    if (roomId !== null && wasEverSaved) {
-      await Promise.all([
-        fetchEquipmentTypes(),
-        fetchRoom(roomId),
-        fetchRoomPermissions(roomId),
-        fetchRoomDetails(roomId, true),
-      ]);
-    }
-    onClose(wasEverSaved || afterCreation);
-  };
-
-  handleSubmit = async (data, form) => {
-    let {roomId} = this.props;
+    return camelizeKeys(response.data);
+  }, []);
+
+  const fetchRoomData = useCallback(async () => {
+    const resp = await Promise.all([
+      fetch(roomURL, {room_id: roomId}),
```

Comment:
I'd call the url builder here instead of passing the params separately, i.e. `roomURL({room_id: roomId})`

Reply:
Sounds good!